### PR TITLE
store target informations to database

### PR DESCRIPTION
**Pull request by LekeneCedric**

### CODE REVIEW

**Comment by LekeneCedric** [comment]
given the management of non-permanent target data, to overcome this problem I integrated the databases so that all target information is stored permanently in a given database

File: main.py [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
 from flask_cors import CORS, cross_origin
 from hashlib import md5
 
-
+    
 app = Flask(__name__)
+
+#initialize configuration
+app.config["SQLALCHEMY_DATABASE_URI"] = "sqlite:///fallamoi.db"
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
```
Comment (Sanix-Darker) [inline main.py:12]: Add these in a configutation file, and import those from there !
it should not be as plain text like that in the codebase !

See this for example : https://gist.github.com/Sanix-Darker/bdaae2dfcb3ab74b356b4c3c799d7844

File: main.py [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
+
+db = SQLAlchemy(app)
+
+#create target database model 
+class Target(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    tid = db.Column(db.String)
+    data = db.Column(db.String,nullable=False)
```
Comment (Sanix-Darker) [inline main.py:20]: This should be a serializable type right ?
not a big fan on saving raw json string as string input 🤔

File: main.py [hunk 3]
```diff
@@ -1,10 +1,25 @@
...
+    #data is a dictionary of data that we will transform into a character string 
+    #but when we want to retrieve them for precise processing, we will parse them in json 
 CORS(app, supports_credentials=False)
```
Comment (Sanix-Darker) [inline main.py:23]: CORS should update the app object before sqlAlchemy !

**Review by Sanix-Darker — COMMENTED** [review]
Thanks for this PR !
The main reason i was saying we should start with a json file called results.json that we we're going to update over the time for new users, is because, the project at this day doesn't require a whole DB ! 🤔

File: main.py [hunk 4]
```diff
@@ -49,29 +64,46 @@ def set_payload(link_uid: str):
...
+    targets = Target.query.all()
+    res = []
+    # foreach target append  her data to array 
+    for target in targets:
+      res.append({
+          'id':target.id,
+          'tid':target.tid,
+          'data':target.data,
+      })
+    return res , 200
```
Comment (Sanix-Darker) [inline main.py:103]:
```suggestion
    return [{ 'id':target.id,  'tid':target.tid, 'data':target.data } for target in Target.query.all()] , 200  
```

File: main.py [hunk 5]
```diff
@@ -49,29 +64,46 @@ def set_payload(link_uid: str):
...
+          'id':target.id,
+          'tid':target.tid,
+          'data':target.data,
+      })
+    return res , 200  
 
 
 if __name__ == '__main__':
+    with app.app_context():
+     db.create_all()
```
Comment (Sanix-Darker) [inline main.py:108]: What are we creating here ? 🤔

File: main.py [hunk 6]
```diff
@@ -49,29 +64,46 @@ def set_payload(link_uid: str):
...
-    return user, 200 if bool(user) else 404
-
+    
+    #get target information
+    target = Target.query.filter_by(tid=user_uid).first()
+    res = {
+          'id':target.id,
+          'tid':target.tid,
+          'data':target.data,
+      }
```
Comment (Sanix-Darker) [inline main.py:88]: Can also be shrink in one or two lines !

File: main.py [hunk 7]
```diff
@@ -49,29 +64,46 @@ def set_payload(link_uid: str):
 @app.route('/save', methods=['POST'])
 def save_info():
     data = request.get_json()
-
     user_key = hashit(json.dumps(data))
+    dat = json.dumps(data)
     USERS[user_key] = data
-
+    #sotre target data to database
+    print(user_key)
```
Comment (Sanix-Darker) [inline main.py:71]:
```suggestion
```

File: main.py [hunk 8]
```diff
@@ -49,29 +64,46 @@ def set_payload(link_uid: str):
 @app.route('/save', methods=['POST'])
 def save_info():
     data = request.get_json()
-
     user_key = hashit(json.dumps(data))
+    dat = json.dumps(data)
```
Comment (Sanix-Darker) [inline main.py:68]:
```suggestion
```

File: main.py [hunk 9]
```diff
@@ -49,29 +64,46 @@ def set_payload(link_uid: str):
...
     USERS[user_key] = data
-
+    #sotre target data to database
+    print(user_key)
+    target = Target(tid = user_key,data = dat)
```
Comment (Sanix-Darker) [inline main.py:72]:
```suggestion
    target = Target(tid = user_key,data = json.dumps(data))
```

**Pull request closed without merging**